### DISCUSSION TITLE
Error in land cover grid

### DISCUSSION THREAD

**Discussion by sanwitiab** [UMEP-dev/UMEP#340] [Q&A] [answered]
Hello,

I try to process the thermal comfort zone. But when I run shows this error "Land cover grid includes Confier land cover class. Ground cover information (underneath canopy) is required."

Please help me. How to fix it? 

Thank you

**Comment by biglimp** [comment, marked as answer]
The land cover in the SOLWEIG model should be ON the ground, i.e. what is underneath the trees. Therefore, the conifer (4) and deciduous (3) class should NOT be present. You should recalss these pixels to e.g. bare soil, paved or grass.

**Comment by protoxide95** [comment]
Good evening, I need help to understand what is the reason for this error when running SOLWEIG:

The land cover grid includes integer values highter (or lower)than UMEP-formattedland cover grid (should be integer between 1 and 7). If other LC-classes should be included they also need to be included in landcoverclasses_2016a.txt (Where can I find this .txt file?)

I have created a land use raster with the same resolution and extent as the other layers by including values from 1 to 7 as specified.

In doing other tests, I also encountered this error:
Error in land cover grid: Land cover grid includes Decidiuous land cover class. Ground cover information (underneath canopy) is required.

How to solve it? Thank you

**Reply by biglimp** [reply]
The text-file is found in The landcoverclasses_2016.txt is found in UMEP/SOLWEIG/ in your plugin directory. Answer to your other error is found above.

**Reply by protoxide95** [reply]
[](url)thank you for your response.
I think I've solved with the land use class talk (3 and 4 not present). The raster created is integer.

I found the landcoverclasses_2016 file again and it has the following strings:

Name Code Alb Emis Ts_deg Tstart TmaxLST
Roofs(buildings) 2 0.18 0.95 0.58 -9.78 15.0
Dark_asphalt 1 0.18 0.95 0.58 -9.78 15.0
Cobble_stone_2014a 0 0.20 0.95 0.37 -3.41 15.0
Water 7 0.05 0.98 0.00 12.0
Grass_unmanaged 5 0.16 0.94 0.21 -3.38 14.0
bare_soil 6 0.25 0.94 0.33 -3.01 14.0
Walls 99 0.20 0.90 0.37 -3.41 15.0

I tried to remove classes 0 (cobble stone) and 99 (walls) which I actually don't have in the raster but the error that comes up is this:

https://ibb.co/G2mVXys

**Reply by biglimp** [reply]
It does not matter how many lines (classes) you have in your file. You can keep 0 and 99 without issues. Your error is probably due to something else. Maybe the wrong separator. Attached your file here so that I can have a look.

**Reply by protoxide95** [reply]
I attach here the file .txt and the landcovergrid (raster)
[landcoverclasses_2016a.txt](https://github.com/UMEP-dev/UMEP/files/11435823/landcoverclasses_2016a.txt)


[landcovertypes_albaadriatica_int_v2.zip](https://github.com/UMEP-dev/UMEP/files/11435840/landcovertypes_albaadriatica_int_v2.zip)

**Reply by taniasab5000** [reply]
I am having the same problem. How can I get rid of no data values?

**Reply by biglimp** [reply]
You both need to get rid of your NoData values. Use the **Fill NoData Cells** found in the Processing Toolbox in QGIS and replace with 1.

**Reply by protoxide95** [reply]
Solved thank you very much. The simulation has started.
Now I have only one problem. I do not have a square/rectangular raster but with a non-regular perimeter, filling in the outer values no date setting 1 (as if it were paved) is a great approximation.
I guess I will have to do the land use outside my perimeter.

**Reply by biglimp** [reply]
or you can just ignore the area where you replaced your Nodata values.

**Reply by protoxide95** [reply]
Will it affect the interior area?

**Reply by biglimp** [reply]
very little, just in the edges of your study area
